IA-730 history log crash on parsing failure

olethanh opened the pull request.

olethanh commented on plugins/polio/api.py:
```diff
@@ -1095,8 +1096,16 @@ def handle_ona_request_with_key(request, key, country_id=None):
+            password=config["password"],
+            minutes=config.get("minutes", 60),
         )
+        if not forms:
```
I would put the try except at this level so it doesn't affect the other place we could use get_url_content  and more info about the error could be included in the e-mail

olethanh commented on plugins/polio/helpers.py:
```diff
@@ -14,15 +15,20 @@ def get_url_content(url, login, password, minutes=60):
+        return j
+    except HTTPError:
+        # Return false in case the WHO server returns an error.
+        return False
```
`logging.exception(<exception>)` should be called so it's properly logged in sentry

olethanh commented on plugins/polio/api.py:
```diff
@@ -1095,8 +1096,16 @@ def handle_ona_request_with_key(request, key, country_id=None):
             continue
         forms = get_url_content(
-            url=config["url"], login=config["login"], password=config["password"], minutes=config.get("minutes", 60)
+            url=config["url"] + "?date_created__month__lte=24",
```
Not too sure about the hard coding of this parameter, is this what you want @madewulf  ?

Also is the parameter actually correct? Month smaller than 24 seems strange

SimonDus replied:
Well, I can add the params in the config. It would be more flexible. 
The last 24 months is set as we discussed with Martin

olethanh reviewed (commented):
@madewulf Can you confirm this is actually the behaviour you expect?

olethanh commented:
I have updated the branch with main  and changed the behavior a bit (removed the filter on the months as it is not relevant anymore), @mathvdh ,  can you take a look?

mathvdh reviewed (approved):
Seems quite straightforward to me ... 
Only question is config["login"] contains the country name then ?
I approved but I'm curious ..

olethanh commented:
the user name for the country on the who server it think
